src/ui/components/filters/filteroptionscomponent.js

```diff
@@ -110,17 +110,36 @@ class FilterOptionsConfig {
         config.previousOptions = [];
       }
     }
-    let selectedOptions = config.previousOptions || [];
-    this.options = this.setDefaultSelectedValues(this.options, selectedOptions);
+    const previousOptions = config.previousOptions || [];
```

Comment:
For my own edification what are previousOptions? Is this what's stored in the URL? If so, it's confusing that this data is coming from the config. Why not query persistent storage or global storage directly?

Reply:
We have this very special block in the constructor that passes "previousOptions" into the config class

```
    let previousOptions = this.core.globalStorage.getState(this.name);
    this.core.globalStorage.delete(this.name);

    /**
     * The component config
     * @type {FilterOptionsConfig}
     */
    this.config = new FilterOptionsConfig({
      previousOptions,
      ...config
    });
```

Reply:
so even though calling it config.previousOptions might make you think it's from the config, it's not! pretty fun

Reply:
Would it make more sense to move this to a separate parameter?

Reply:
That's pretty weird. Don't worry about it for now, this was before your PR! Let's just make an item to address.

Reply:
but how can I scope creep the entire codebase into this then